cli/pack/common.go

```diff
@@ -113,6 +115,16 @@ func skipArtifacts(cliOpts *config.CliOpts) func(src string) (bool, error) {
 func prepareBundle(cmdCtx *cmdcontext.CmdCtx, packCtx *PackCtx,
 	cliOpts *config.CliOpts, buildRocks bool) (string, error) {
 	var err error
+	var signer integrity.Signer = nil
+
+	// If integrity checks are enabled, create an IntegritySigner.
+	if packCtx.IntegrityPrivateKey != "" {
+		signer, err = integrity.NewSigner(packCtx.IntegrityPrivateKey)
+
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
Otherwise we need to create some kind of a dummy implementation any way.
```suggestion
	signer, err := NewSigner(packCtx.IntegrityPrivateKey)}
	if err != nil {
		return "", err	
	}
```